src/services/utilities.ts

```diff
@@ -3273,23 +3273,62 @@ namespace ts {
         }
     }
 
-    export function createModuleSpecifierCache(): ModuleSpecifierCache {
-        let cache: ESMap<Path, boolean | readonly ModulePath[]> | undefined;
+    export interface ModuleSpecifierResolutionCacheHost {
+        watchNodeModulesForPackageJsonChanges(directoryPath: string): FileWatcher;
+    }
+
+    export function createModuleSpecifierCache(host: ModuleSpecifierResolutionCacheHost): ModuleSpecifierCache {
+        let containedNodeModulesWatchers: ESMap<string, FileWatcher> | undefined;
+        let cache: ESMap<Path, boolean | { modulePaths: readonly ModulePath[], moduleSpecifiers?: readonly string[] }> | undefined;

```

Comment:
can you please add comment on boolean and object values so its clear what it means since this cache is overloaded with  result..
Why should it not be `boolean | readonly ModulePath[] | { modulePaths: readonly ModulePath[], moduleSpecifiers: readonly string[]}`